File: app/src/main/java/edu/tjrac/swant/douban/view/DoubanMovieFragment.java
package edu.tjrac.swant.douban.view;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.CoordinatorLayout;
import android.support.v4.widget.NestedScrollView;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.yckj.baselib.common.base.BaseMVPFragment;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import edu.tjrac.swant.douban.adapter.DoubanMoviesAdapter;
import edu.tjrac.swant.douban.contract.DoubanMovieContract;
import edu.tjrac.swant.douban.doubane.DoubanCommingMovieBean;
import edu.tjrac.swant.douban.doubane.DoubanIsShowingBean;
import edu.tjrac.swant.douban.doubane.MovieData;
import edu.tjrac.swant.douban.presenterImpl.DoubanMoviePresenter;
import edu.tjrac.swant.unicorn.R;
import edu.tjrac.swant.unicorn.view.SearchActivity;
import edu.tjrac.swant.zhihu.zhihu.SubjectsBean;

/**
 * 类描述:
 * 创建人: Administrator
 * 创建时间: 2018/3/17 0017 上午 9:42
 * 修改人:
 * 修改时间:
 * 修改备注:
 */

public class DoubanMovieFragment extends BaseMVPFragment<DoubanMoviePresenter> implements DoubanMovieContract.View {

    @BindView(R.id.recycler) RecyclerView mRecycler;
    @BindView(R.id.recycler2) RecyclerView mRecycler2;
    @BindView(R.id.recycler3) RecyclerView mRecycler3;
    @BindView(R.id.recycler4) RecyclerView mRecycler4;

    @BindView(R.id.cl) CoordinatorLayout cl;
    @BindView(R.id.nsv) NestedScrollView nsv;

//    @BindView(R.id.swiper) SwipeRefreshLayout mSwiper;

    DoubanMoviesAdapter adapter,adapter2,adapter3,adapter4;

    ArrayList<MovieData> isShowing = new ArrayList<>();
    ArrayList<MovieData> comming = new ArrayList<>();
    ArrayList<MovieData> top250 = new ArrayList<>();
    ArrayList<MovieData> mb = new ArrayList<>();
@BindView(R.id.iv_head) ImageView head;

    Unbinder unbinder;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_douban_movie, container, false);

        unbinder = ButterKnife.bind(this, v);

        adapter = new DoubanMoviesAdapter(isShowing);
        adapter2=new DoubanMoviesAdapter(comming);
        adapter3 = new DoubanMoviesAdapter(top250);
        adapter4=new DoubanMoviesAdapter(mb);

        mRecycler.setLayoutManager(new LinearLayoutManager(getActivity(),LinearLayoutManager.HORIZONTAL,false));
        mRecycler2.setLayoutManager(new LinearLayoutManager(getActivity(),LinearLayoutManager.HORIZONTAL,false));
        mRecycler3.setLayoutManager(new LinearLayoutManager(getActivity(),LinearLayoutManager.HORIZONTAL,false));
        mRecycler4.setLayoutManager(new LinearLayoutManager(getActivity(),LinearLayoutManager.HORIZONTAL,false));

        mRecycler.setAdapter(adapter);
        mRecycler2.setAdapter(adapter2);
        mRecycler3.setAdapter(adapter3);
        mRecycler4.setAdapter(adapter4);

        adapter.setOnItemClickListener(new BaseQuickAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(BaseQuickAdapter adapter, View view, int position) {
//                DoubanMovieDetialActivity.start(getActivity(), (SubjectsBean) movies.get(position));
                ImageView imageView = view.findViewById(R.id.image);
                DoubanMovieDetialActivity.startWithSharedView(getActivity(), imageView, (SubjectsBean) isShowing.get(position));
            }
        });

        presenter = new DoubanMoviePresenter(this);

        presenter.getIsShowingMovies();
        presenter.getCommingMovies();

//        mSwiper.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
//            @Override
//            public void onRefresh() {
//                isShowing.clear();
//                comming.clear();
//                presenter.getIsShowingMovies();
//                presenter.getCommingMovies();
//            }
//        });

        setHasOptionsMenu(true);
        cl.setOnHierarchyChangeListener(new ViewGroup.OnHierarchyChangeListener() {
            @Override
            public void onChildViewAdded(View parent, View child) {

            }

            @Override
            public void onChildViewRemoved(View parent, View child) {

            }
        });
        head.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                CoordinglayoutTestActivity.start(getActivity());
            }
        });
        return v;
    }

    @Override
    public void onBack() {

    }

    @Override
    public void onGetIsShowingSuccess(DoubanIsShowingBean bean) {
        isShowing.addAll(bean.getSubjects());
        adapter.notifyDataSetChanged();

    }

    @Override
    public void showProgress() {
        super.showProgress();
//        mSwiper.setRefreshing(true);
    }

    @Override
    public void dismissProgress() {
        super.dismissProgress();
//        if (mSwiper.isRefreshing()) {
//            mSwiper.setRefreshing(false);
//        }
    }

    @Override
    public void onGetCommingSuccess(DoubanCommingMovieBean bean) {
        comming.addAll(bean.getSubjects());
        adapter2.notifyDataSetChanged();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }




    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        inflater.inflate(R.menu.douban_menu, menu);

        super.onCreateOptionsMenu(menu, inflater);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.seach:
                SearchActivity.start(getActivity(),2);
                break;
//            case R.id.top250:
//                LeaderboardActivity.start(getActivity(), "top250");
//                break;
//
//            case R.id.us_box:
//                LeaderboardActivity.start(getActivity(), "us_box");
//                break;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/edu/tjrac/swant/douban/view/DoubanMovieFragment.kt
package edu.tjrac.swant.douban.view

import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.view.*
import android.widget.ImageView
import com.yckj.baselib.common.base.BaseMVPFragment
import edu.tjrac.swant.douban.adapter.DoubanMoviesAdapter
import edu.tjrac.swant.douban.contract.DoubanMovieContract
import edu.tjrac.swant.douban.doubane.DoubanCommingMovieBean
import edu.tjrac.swant.douban.doubane.DoubanIsShowingBean
import edu.tjrac.swant.douban.doubane.MovieData
import edu.tjrac.swant.douban.presenterImpl.DoubanMoviePresenter
import edu.tjrac.swant.unicorn.R
import edu.tjrac.swant.unicorn.view.SearchActivity
import edu.tjrac.swant.zhihu.zhihu.SubjectsBean
import kotlinx.android.synthetic.main.fragment_douban_movie.*
import java.util.*

/**
 * 类描述:
 * 创建人: Administrator
 * 创建时间: 2018/3/17 0017 上午 9:42
 * 修改人:
 * 修改时间:
 * 修改备注:
 */

class DoubanMovieFragment : BaseMVPFragment<DoubanMoviePresenter>(), DoubanMovieContract.View {

    //    @BindView(R.id.recycler) RecyclerView recycler;
    //    @BindView(R.id.recycler2) RecyclerView recycler2;
    //    @BindView(R.id.recycler3) RecyclerView recycler3;
    //    @BindView(R.id.recycler4) RecyclerView recycler4;

    //    @BindView(R.id.cl) CoordinatorLayout cl;
    //    @BindView(R.id.nsv) NestedScrollView nsv;

    //    @BindView(R.id.swiper) SwipeRefreshLayout mSwiper;

    internal var adapter: DoubanMoviesAdapter?=null
    internal var adapter2: DoubanMoviesAdapter?=null
    internal var adapter3: DoubanMoviesAdapter?=null
    internal var adapter4: DoubanMoviesAdapter?=null

    internal var isShowing = ArrayList<MovieData>()
    internal var comming = ArrayList<MovieData>()
    internal var top250 = ArrayList<MovieData>()
    internal var mb = ArrayList<MovieData>()
    //@BindView(R.id.iv_iv_head) ImageView iv_head;

    //    Unbinder unbinder;

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val v = inflater.inflate(R.layout.fragment_douban_movie, container, false)


        adapter = DoubanMoviesAdapter(isShowing)
        adapter2 = DoubanMoviesAdapter(comming)
        adapter3 = DoubanMoviesAdapter(top250)
        adapter4 = DoubanMoviesAdapter(mb)

        recycler.setLayoutManager(LinearLayoutManager(activity, LinearLayoutManager.HORIZONTAL, false))
        recycler2.setLayoutManager(LinearLayoutManager(activity, LinearLayoutManager.HORIZONTAL, false))
        recycler3.setLayoutManager(LinearLayoutManager(activity, LinearLayoutManager.HORIZONTAL, false))
        recycler4.setLayoutManager(LinearLayoutManager(activity, LinearLayoutManager.HORIZONTAL, false))

        recycler.setAdapter(adapter)
        recycler2.setAdapter(adapter2)
        recycler3.setAdapter(adapter3)
        recycler4.setAdapter(adapter4)

        adapter!!.setOnItemClickListener { adapter, view, position ->
            //                DoubanMovieDetialActivity.start(getActivity(), (SubjectsBean) movies.get(position));
            val imageView = view.findViewById<ImageView>(R.id.image)
            DoubanMovieDetialActivity.startWithSharedView(activity!!, imageView, isShowing[position] as SubjectsBean)
        }

        presenter = DoubanMoviePresenter(this)

        presenter.getIsShowingMovies()
        presenter.getCommingMovies()

        //        mSwiper.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
        //            @Override
        //            public void onRefresh() {
        //                isShowing.clear();
        //                comming.clear();
        //                presenter.getIsShowingMovies();
        //                presenter.getCommingMovies();
        //            }
        //        });

        setHasOptionsMenu(true)
        cl.setOnHierarchyChangeListener(object : ViewGroup.OnHierarchyChangeListener {
            override fun onChildViewAdded(parent: View, child: View) {

            }

            override fun onChildViewRemoved(parent: View, child: View) {

            }
        })
        iv_head.setOnClickListener(View.OnClickListener {
            //                CoordinglayoutTestActivity.start(getActivity());
        })
        return v
    }

    override fun onBack() {

    }

    override fun onGetIsShowingSuccess(bean: DoubanIsShowingBean) {
        isShowing.addAll(bean.subjects)
        adapter!!.notifyDataSetChanged()

    }

    override fun showProgress() {
        super.showProgress()
        //        mSwiper.setRefreshing(true);
    }

    override fun dismissProgress() {
        super.dismissProgress()
        //        if (mSwiper.isRefreshing()) {
        //            mSwiper.setRefreshing(false);
        //        }
    }

    override fun onGetCommingSuccess(bean: DoubanCommingMovieBean) {
        comming.addAll(bean.subjects)
        adapter2!!.notifyDataSetChanged()
    }

    override fun onDestroyView() {
        super.onDestroyView()
    }


    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
        inflater!!.inflate(R.menu.douban_menu, menu)

        super.onCreateOptionsMenu(menu, inflater)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item!!.itemId) {
            R.id.seach -> SearchActivity.start(activity, 2)
        }//            case R.id.top250:
        //                LeaderboardActivity.start(getActivity(), "top250");
        //                break;
        //
        //            case R.id.us_box:
        //                LeaderboardActivity.start(getActivity(), "us_box");
        //                break;
        return super.onOptionsItemSelected(item)
    }
}
